lessons/configuration.md

```diff
@@ -28,15 +28,36 @@ This lessons consists of the following exercises:
 
 ## 1. Why do we use configuration? (Stace)
 
-### Steps
+When we run our applications locally, we run them against a known environment. We know the location of all resources we need, and if the code is staying on our machines then we also know all of our secrets.
 
-1.
-2.
-3.
+During the 90's, when deploying small applications you could even take a good guess that the users running your machine would have a similar environment to your own.
 
-> 📝 **Tip** - < TIP >
+If they were running DOS you could assume that they had a C drive where you could store data. Copy your file to the machines that need to run the application and it just works.
 
-> 🔎 **Observation** - < OBSERVATION >
+That didn't work for larger applications though, and in this day and age it doesn't really work at all.
+
+As applications became larger, as they became more complex and distributed, and as they were deployed to more diverse environments, you - the developer - lost the knowledge of where and how the application was going to run.
+
+You could make a change to your application and recompile it for each environment that you deploy into - but that is both a lot of extra (manual) work, and introduces risks as what is deployed to test and production environments are all different applications. This is not a realistic approach.
+
+The solution to this is to introduce configuration settings to your application. Something that can be changed externally to your code to allow the same runtime to work in multiple different places. That is, use environment settings to separate the runtime values in your code from the compile time.
```

Comment:
Let's stick to one name for 'configuration settings'. I suggest we use _application settings_ or _app settings_ since this aligns closely with MS/Azure naming. App settings is a form of configuration. How does that sound?